contracts/marketupdates/MarketUpdateTimelock.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.10;
+
+import "./SafeMath.sol";
+
+/*
+
+Right now admin and marketUpdateProposer can cancel one another's transactions, but
+this is not a realistic scenario as admin is a main-governor-timelock which will not be
+queuing, executing, or cancelling transactions. So we are not handling or testing it.
+*/
+contract MarketUpdateTimelock {
+    using SafeMath for uint;
+
+    event NewAdmin(address indexed oldAdmin, address indexed newAdmin);
+    event NewMarketUpdateProposer(address indexed oldMarketAdmin, address indexed newMarketAdmin);
+    event NewDelay(uint indexed newDelay);
+    event CancelTransaction(bytes32 indexed txHash, address indexed target, uint value, string signature,  bytes data, uint eta);
+    event ExecuteTransaction(bytes32 indexed txHash, address indexed target, uint value, string signature,  bytes data, uint eta);
+    event QueueTransaction(bytes32 indexed txHash, address indexed target, uint value, string signature, bytes data, uint eta);
+
+    uint public constant GRACE_PERIOD = 14 days;
+    uint public constant MINIMUM_DELAY = 0 days;
```

Comment:
This should not be `0`. Added currently to make the testing easy. Will update it